src/ucs/sys/sys.c

```diff
@@ -314,14 +314,15 @@ ssize_t ucs_read_file(char *buffer, size_t max, int silent,
     return read_bytes;
 }
 
-static unsigned long ucs_sysconf(int name)
+static long ucs_sysconf(int name)
 {
     long rc;
+    errno = 0;
 
     rc = sysconf(name);
-    ucs_assert_always(rc >= 0);
+    ucs_assert_always(errno == 0);
 
-    return (unsigned long)rc;
```

Comment:
users of ucs_sysconf() do not expect a negative value. need to make it `static unsigned long ucs_sysconf(int name, unsigned long default)`, so the default value would be returned in case of "-1"
This is also the reason for Coverity failure

Reply:
And this is bug as well. Users should be able to deal with -1. There is no such thing as a default value for env.

Reply:
1. need to assert_always that the rc is either positive or -1
1. need to document that if -1 is returned form this function, it means "undefined"
1. need to update ucs_get_max_iov() as well to handle -1.

Reply:
disagree with 1. 
rc can be any value (negative, zero, non zero) and it is all legal. There is nothing to assert there. What we should assert here is errno. this is it.